Merchant/App/Scanner/ScannerViewController.swift

```diff
@@ -135,9 +131,9 @@ class ScannerViewController: UIViewController, AVCaptureMetadataOutputObjectsDel
         captureSession = nil
     }
     
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-        setupCamera()
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        checkPermission()
```

Comment:
Cf. Life cycle of a view controller. ViewDidLoad will be triggered without the ViewDidAppear. No change required.

Reply:
Could you please elaborate more on this. I'm not sure what do you mean. `viewDidLoad` > `viewWillAppear` > `viewDidAppear`.